pkg/internal/meshdetect/reports_mesh_test.go

```diff
@@ -0,0 +1,160 @@
+package meshdetect
+
+import (
+	"testing"
+)
+
+func TestSerializeMeshDeploymentResults(t *testing.T) {
+	testCases := []struct {
+		caseName      string
+		results       map[MeshKind]*DeploymentResults
+		serializedStr string
+	}{
+		{
+			caseName: "deployment:kong-mesh",
+			results: map[MeshKind]*DeploymentResults{
+				MeshKindKongMesh: {
+					ServiceExists: true,
+				},
+			},
+			serializedStr: "mdep=\"km3\"",
+		},
+		{
+			caseName: "deployment:traefik",
+			results: map[MeshKind]*DeploymentResults{
+				MeshKindTraefik: {
+					ServiceExists: true,
+				},
+			},
+			serializedStr: "mdep=\"t3\"",
+		},
+		{
+			caseName: "deployment:consul,aws-app-mesh",
+			results: map[MeshKind]*DeploymentResults{
+				MeshKindConsul: {
+					ServiceExists: true,
+				},
+				MeshKindAWSAppMesh: {
+					ServiceExists: true,
+				},
+			},
+			serializedStr: "mdep=\"a3,c3\"",
+		},
+		{
+			caseName:      "deployment:nil results should produce empty string",
+			results:       nil,
+			serializedStr: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		// t.Run runs function in separate goroutine, so we need to assign tc to a local variable.
+		tc := tc
+		t.Run(tc.caseName, func(t *testing.T) {
+			// serialized := serializeMeshDeploymentResults(tc.results)
+			// require.Equalf(t, tc.serializedStr, serialized,
+			// 	"case %s: serialized message should be the same as expected", tc.caseName)
```

Comment:
We're not running any test here, is that correct? 🤔

Reply:
Yikes! 😬 Fixed.